### PR TITLE
Create separate category pages, separate "all articles" index

### PR DESCRIPTION
**Pull request by zachmargolis**
**Problem**: The Handbook is kind of an unordered mess of articles, it could use some information architecture and grouping. 

**Proposal**:

- Create formal "category" pages so that we can choose to add more curated content on a per-category basis (note: I didn't actually create any of that content... but we *could*!)
- Remove the old `highlight_categories.yml` it was a cute bandaid for custom ordering but hard to maintain

**Result**:
So this is more of a "clean up the homepage" kind of thing, and it creates `/all` that still preserves the "link to everything" page. Would love ideas on exemplar category pages

**Future Ideas**:
- Maybe we clean up/nest/combine some categories (Architecture is kind of under Development) and add sections there to curate things?

| before | after |
| --- | --- |
|  Screen Shot 2022-08-18 at 5 21 48 PM |  Screen Shot 2022-08-18 at 5 21 52 PM |
|  N/A | Screen Shot 2022-08-18 at 5 23 35 PM |
| N/A |  Screen Shot 2022-08-18 at 5 26 39 PM |

### CODE REVIEW

File: _categories/appdev.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+---
+title: AppDev
+layout: category
+---
```
Comment (zachmargolis) [inline _categories/appdev.md:4]: we could also consider manual ordering of the cards and things with like an `order: 1` style thing

**Comment by zachmargolis** [comment]
🆕 a few updates

| change | screenshot |
| ---- | ---- | 
| Remove `/all` and bring everything back to the homepage | N/A |
| category icons using USWDS | Screen Shot 2022-08-19 at 10 38 04 AM |
| new `subcategory` field that lets us group articles | Screen Shot 2022-08-19 at 10 38 14 AM |

File: index.md [hunk 2]
```diff
@@ -11,25 +11,38 @@ to [avoid contributing sensitive information][sensitive-information].
...
+  {% for category in site.categories %}
+    <li class="usa-card tablet:grid-col-4">
+      <div class="usa-card__container">
+        <div class="usa-card__header">
+          <span class="usa-card__heading">
+            {% if category.icon %}
+              {% include uswds-icon.html icon=category.icon %}
+            {% endif %}
+            <strong>
+              <a href="{{ category.url | prepend: site.baseurl }}">{{ category.title }}</a>
```
Comment (aduth) [inline index.md:26]: Can we add design system link styling?

```suggestion
              <a href="{{ category.url | prepend: site.baseurl }}" class="usa-link">{{ category.title }}</a>
```

File: index.md [hunk 3]
```diff
@@ -11,25 +11,38 @@ to [avoid contributing sensitive information][sensitive-information].
...
-{%     include articles_category.md
-               category=category
-               articles=site.articles %}
-{%   endunless %}
-{% endfor %}
+## Categories
+
+<ul class="usa-card-group">
+  {% for category in site.categories %}
+    <li class="usa-card tablet:grid-col-4">
```
Comment (aduth) [inline index.md:18]: The text wrapping that happens with the icons is kinda an eye sore.

Screen Shot 2022-08-19 at 2 29 38 PM

What do you think about two-column instead?

```suggestion
    <li class="usa-card tablet:grid-col-6">
```

File: _articles/eng-design-doc.md [hunk 4]
```diff
@@ -3,6 +3,6 @@ title: "Engineering Design Doc Template"
 layout: article
 category: "AppDev"
 redirect_to: https://docs.google.com/document/d/1KlQL4MMCXahV12mMbh5scmXLl6adjv_0fcFzQ8si7qc/edit#
-description: "Template that can be copied for engineering proposals, replaced in favor of [RFCs](#rfcs)"
+description: "Template that can be copied for engineering proposals, replaced in favor of [RFCs](/categories/team.html#rfcs)"
```
Comment (aduth) [inline _articles/eng-design-doc.md:6]: Is it possible to get `baseurl` in here to make sure it works in Federalist previews?

File: index.md [hunk 5]
```diff
@@ -11,25 +11,38 @@ to [avoid contributing sensitive information][sensitive-information].
...
-             custom_order=category.custom_order %}
-{% endfor %}
-{% for category in all_categories %}
-{%   unless highlight_categories_titles contains category %}
-{%     include articles_category.md
-               category=category
-               articles=site.articles %}
-{%   endunless %}
-{% endfor %}
+## Categories
```
Comment (aduth) [inline index.md:14]: What would you think about adding a bit of space here? Frustratingly both the heading and the `.usa-card-group` aren't very flexible with adding margin utility classes, but we could add a `<div>` or `<div>` wrapper.

```suggestion
## Categories

<div class="margin-bottom-4"></div>
```

**Review by aduth — APPROVED** [review]
LGTM! 👍

File: _articles/eng-design-doc.md (hunk 4 above)
Comment (zachmargolis) [reply _articles/eng-design-doc.md:6]: I don't think that YAML frontmatter gets expanded out :[

Comment (zachmargolis) [reply _articles/eng-design-doc.md:6]: I'll try it just to see

Comment (zachmargolis) [reply _articles/eng-design-doc.md]: c2be93c81d5fdfc6e9f051b319899a4755c02ae3

Comment (zachmargolis) [reply _articles/eng-design-doc.md]: Confirmed, it does not get expanded out. [HTMLproofer errored in CI](https://app.circleci.com/pipelines/github/18F/identity-handbook/961/workflows/89b269e1-4e48-4803-9eef-dd4d2fa6df2b/jobs/962) on this

```
- /home/circleci/identity-handbook/_site/index.html
  *  internally linking to {{ site.baseurl }}/categories/team.html#rfcs, which does not exist (line 404)
     <a class="usa-link" href="{{ site.baseurl }}/categories/team.html#rfcs">RFCs</a>
```

**Pull request merged**